### DISCUSSION TITLE
Awesome technical stacks with this project!

### DISCUSSION THREAD

**Discussion by TaylorDurden** [hongbo-miao/hongbomiao.com#2739] [Q&A] [answered]
Do you have a plan to give us a course or documentation to demonstrate all the good technical stacks with this project?

**Comment by hongbo-miao** [comment, marked as answer]
Hey @TaylorDurden thanks for the word!

This repo is currently being used by me to experiment cutting-edge technologies which keeps changing.

If you want to learn from this repo, I would suggest first checking technologies listed in the [Tech Stack](https://github.com/Hongbo-Miao/hongbomiao.com#-tech-stack) section, and go these technology repos to learn how to use them separately. For some big technologies such as Kafka, there are many very high-quality courses on Udemy, other learning platforms, or books for them. I learned from those places too.

More importantly, focus on the concept and what problem the technology is trying to solve, instead of the technology itself. Because technologies come and go very fast, from time to time, there are always better alternatives, but the concept almost never changes.

For example, in the Tech Stack list, `Opossum` is the library I am using for circuit breaker in a Node.js server. But in Go, Python, or Java projects, they have their own libraries for circuit breaker. Also, for service mesh, such as Istio, it provides circuit breaker function between all its microservices.

So instead of focusing on `Opossum` library itself, focus on learning what is circuit breaker, and why people introduce this concept. After learning the concept, you know when you need a circuit breaker. If you are building a Go project for example, you can find a Go circuit breaker library, and try to integrate with your project and experimenting.

Hope it helps! 😀
